r/vignettes/install.Rmd

```diff
@@ -284,7 +284,11 @@ so that we can attempt to improve the script.
 ## Known installation issues
 
 * On CentOS, if you are using a more modern `devtoolset`, you may need to set
-the environment variables `CC=/usr/bin/gcc CXX=/usr/bin/g++`.
+the environment variables `CC` and `CXX` (e.g., in the R `Makeconf` file).
+These may be set to system defaults (e.g., `CC=/usr/bin/gcc CXX=/usr/bin/g++`)
+or they may be set to use the `gcc`/`g++` from the devtoolset. In the latter
+case, you will have to make sure that the system arrow dependencies are not
+used (i.e., make sure the system variable `ARROW_USE_PKG_CONFIG` is false).
```

Comment:
... "because the system packages are built with the system compilers"?

Is that the reason? It's worth explaining IMO.

Interestingly, for CentOS 6 we do build with devtoolset-8 (https://github.com/apache/arrow/blob/master/dev/tasks/linux-packages/apache-arrow/yum/centos-6/Dockerfile#L23), but we don't for CentOS 7 or higher. @kou does that make sense?

Reply:
Yep, just added more explanation. In my testing I got back to the segfault on `write_parquet(...)` using the devtoolset-8 to compile the R package but using arrow-devel/parquet-devel from the system (and compiled by the system compilers)

Reply:
👍 thanks!

Reply:
Yes.
We use devtoolset only on CentOS 6 because the system GCC on CentOS 6 is too old to build Apache Arrow C++. CentOS 7 or higher use the system GCC because they aren't old to build Apache Arrow C++.

FYI: I'll dropped support for CentOS 6 soon because CentOS 6 reaches EOL.